examples/plot_partial_dependence.py

```diff
@@ -52,48 +64,55 @@
 
 from mpl_toolkits.mplot3d import Axes3D
 
-from sklearn.model_selection import train_test_split
 from sklearn.ensemble import GradientBoostingRegressor
-from sklearn.ensemble.partial_dependence import plot_partial_dependence
-from sklearn.ensemble.partial_dependence import partial_dependence
+from sklearn.neural_network import MLPRegressor
+from sklearn.partial_dependence import plot_partial_dependence
+from sklearn.partial_dependence import partial_dependence
 from sklearn.datasets.california_housing import fetch_california_housing
 
 
 def main():
     cal_housing = fetch_california_housing()
 
-    # split 80/20 train-test
-    X_train, X_test, y_train, y_test = train_test_split(cal_housing.data,
-                                                        cal_housing.target,
-                                                        test_size=0.2,
-                                                        random_state=1)
+    X, y = cal_housing.data, cal_housing.target
     names = cal_housing.feature_names
 
+    # Center target to avoid GBDT init bias: GBDT with 'recursion' method does
+    # not account for the initial estimator (here the average target)
+    y -= y.mean()
+
+    print("Training MLPRegressor...")
+    est = MLPRegressor(activation='logistic')
+    est.fit(X, y)
+    print('Computing partial dependence plots...')
+    features = [0, 5, 1, 2]
```

Comment:
why do we not do (5, 1) here? It makes it seem like this can't be done.

Reply:
It can be done, but it's significantly slower: complexity is squared and the 'brute' method is not optimized so it takes time. I'll add a comment.

Reply:
Thanks for the comment. By saying "the 'brute' method is not optimised" you're not implying that there's a way to make it much faster in the general case, but there might be estimator-specific improvements?